### DISCUSSION TITLE
How is the performance of a dozen or several dozen WebViews in a single ToolWindow?

### DISCUSSION THREAD

**Discussion by JakeRadMSFT** [MicrosoftEdge/WebView2Feedback#3896] [Q&A]
Hello!
My team is working on building a WPF Tool that alternates:

Text Block
WebView
Text Block
WebView
....
Text Block
WebView
Text Block
WebView

How well would you expect 12-15 WebViews in a single ToolWindow to work? What about 100?

We don't really need/want each of those WebViews to be isolated - ideally they would be in the same "Browser" but just different Divs being displayed. If this would somehow help perf ...

Similarly - should we be looking at anything in particular for virtualizing these? Has anyone done this - so they only get rendered when they're in-view?

**Comment by champnic** [comment]
Hey Jake! I'm not sure I've ever tried that many WebView2's in a single window before (let alone 100!). The process model for WebView2 is described here: https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/process-model
Essentially most processes would be shared, but you might end up with a new renderer process per WebView2. We are looking at allowing apps to specify when renderer processes can be shared in the future.

I would definitely suggest trying to minimize the number of WebView2 instances if perf is a concern though - can the TextBlocks be displayed in the WebView2 instead? You might also be able to make the WebView2 transparent and show TextBlocks behind it, but you won't be able to click the TextBlocks.

Thanks!
